File: game-utils/match-viewer/src/match-viewer/math/sineEquation.ts
/**
 * A partial sine equation curveRef structure in format A + r * sin(B * (t - t0) + C).
 *
 * A - offsets in y axis
 * r - radius of the function
 * B - frequency of the sine
 * t0 - time when the equation started
 * C - offset in x axis
 */
export class SineEquation {
    t0: number;
    A: number;
    r: number;
    B: number;
    C: number;

    constructor(t0: number, A: number, r: number, B: number, C: number) {
        this.t0 = t0;
        this.A = A;
        this.r = r;
        this.B = B;
        this.C = C;
    }

    // Returns a value on sine function.
    //
    // time - time based on the start of the function (time at start
    //        of the function is 0)
    getValue(time: number): number {
        return this.A + this.r * Math.sin(this.B * (time - this.t0) + this.C);
    }
}

File: games/planetization/match-generator/core/src/com/planet_lia/match_generator/libs/math/SineEquation.java
package com.planet_lia.match_generator.libs.math;

public class SineEquation {
    public float t0;
    public float A;
    public float r;
    public float B;
    public float C;

    public SineEquation(float t0, float a, float r, float b, float c) {
        this.t0 = t0;
        A = a;
        this.r = r;
        B = b;
        C = c;
    }

    public float getValue(float time) {
        return (float) (this.A + this.r * Math.sin(this.B * (time - this.t0) + this.C));
    }
}
